share/new_eds/rsmt2d.go

```diff
@@ -0,0 +1,116 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ Accessor = Rsmt2D{}
```

Comment:
That's what cool kids do, but we are not doing it in our codebase. Not sure it is worth it as our tests would anyway catch if the implementation doesn't follow the interface anymore

Reply:
It is nice to have safety feature, that is free. Also it will be there of all other Accessor implementations

Reply:
It is useful to have this check to catch problems on compile time (or even by ide), instead of waiting for test runs